test/E2ETest/Microsoft.Test.E2E.AspNet.OData/Build.AspNet/BulkOperation/BulkOperationPatchHandlers.cs

```diff
@@ -0,0 +1,620 @@
+        {
+            PatchStatus status = PatchStatus.Success;
+            errorMessage = string.Empty;
+            originalObject = null;
+
+            try
+            {
+                var id = keyValues["Id"].ToString();
+                originalObject = employee.Friends.FirstOrDefault(x => x.Id == Int32.Parse(id));
+
+
+                if (originalObject == null)
+                {
+                    status = PatchStatus.NotFound;
+                }
+
+            }
+            catch (Exception ex)
+            {
+                status = PatchStatus.Failure;
+                errorMessage = ex.Message;
+            }
+
+            return status;
+        }
+
+        public override IPatchMethodHandler GetNestedPatchHandler(Friend parent, string navigationPropertyName)
+        {
+            switch (navigationPropertyName)
+            {
+                case "Orders":
+                    return new OrderPatchHandler(parent);
+                default:
+                    return null;
+
+            }
+        }
+
+    }
+
+
+    public class OrderPatchHandler : PatchMethodHandler<Order>
+    {
+        Friend friend;
+        public OrderPatchHandler(Friend friend)
+        {
+            this.friend = friend;
+        }
+
+        public override PatchStatus TryCreate(Delta<Order> patchObject, out Order createdObject, out string errorMessage)
+        {
+            createdObject = null;
+            errorMessage = string.Empty;
+
+            try
+            {
+                createdObject = new Order();
+                
+                if(friend.Orders == null)
+                {
+                    friend.Orders = new List<Order>();
+                }
+
+                friend.Orders.Add(createdObject);
+
+                return PatchStatus.Success;
+            }
+            catch (Exception ex)
+            {
+                errorMessage = ex.Message;
+
+                return PatchStatus.Failure;
+            }
+        }
+
+        public override PatchStatus TryDelete(IDictionary<string, object> keyValues, out string errorMessage)
+        {
+            errorMessage = string.Empty;
+
+            try
+            {
+                var id = keyValues.First().Value.ToString();
+                var friend = this.friend.Orders.FirstOrDefault(x => x.Id == int.Parse(id));
+
+                this.friend.Orders.Remove(friend);
+
+                return PatchStatus.Success;
+            }
+            catch (Exception ex)
+            {
+                errorMessage = ex.Message;
+
+                return PatchStatus.Failure;
+            }
+        }
+
+        public override PatchStatus TryGet(IDictionary<string, object> keyValues, out Order originalObject, out string errorMessage)
+        {
+            PatchStatus status = PatchStatus.Success;
+            errorMessage = string.Empty;
+            originalObject = null;
+
+            try
```

Comment:
why put BulkOperationPatchHandlers.cs in BulkOperation folder, and such folder under "Build.AspNet" ?

It's only for ASP.NET Classic?

Reply:
Yes, this is only for classic, we have a seperate one for core, which uses EF

---
In reply to: [604626228](https://github.com/OData/WebApi/pull/2455#discussion_r604626228) [](ancestors = 604626228)